torch/csrc/Size.cpp

```diff
@@ -23,10 +23,18 @@ static PyObject * THPSize_pynew(PyTypeObject *type, PyObject *args, PyObject *kw
   if (self) {
     for (Py_ssize_t i = 0; i < PyTuple_Size(self); ++i) {
       PyObject *item = PyTuple_GET_ITEM(self.get(), i);
-      if (!THPUtils_checkLong(item)) {
-        return PyErr_Format(PyExc_TypeError, "torch.Size() takes an iterable of 'int' (item %zd is '%s')",
-            i, Py_TYPE(item)->tp_name);
+      if (THPUtils_checkLong(item)) {
+        continue;
       }
+      // Try replacing item with the result of item.__index__()
+      THPObjectPtr index(PyObject_CallMethod(item, "__index__", nullptr));
```

Comment:
The result of calling `__index__` isn't really an index, right? :)

Reply:
Btw, I think you want to use `PyNumber_Index` instead of looking up the method